src/torrent.cpp

```diff
@@ -6102,6 +6501,37 @@ bool is_downloading_state(int const st)
 					ret.piece_priorities.push_back(m_picker->piece_priority(i));
 			}
 		}
+
+		if (m_torrent_file->info_hash().has_v2())
+		{
+			ret.merkle_trees.clear();
+			ret.merkle_trees.reserve(m_torrent_file->merkle_trees().size());
+			for (auto const& t : m_torrent_file->merkle_trees())
+				ret.merkle_trees.emplace_back(t.begin(), t.end());
+			if (has_hash_picker())
+			{
+				auto const& leafs = get_hash_picker().verified_leafs();
+				ret.verified_leaf_hashes.clear();
+				ret.verified_leaf_hashes.reserve(leafs.size());
+				for (auto const& l : leafs)
+					ret.verified_leaf_hashes.emplace_back(l);
+			}
+			else if (!m_have_all)
+			{
+				ret.verified_leaf_hashes.reserve(m_torrent_file->files().num_files());
+				for (file_index_t f(0); f != m_torrent_file->files().end_file(); ++f)
+				{
+					if (m_torrent_file->files().pad_file_at(f))
+					{
+						ret.verified_leaf_hashes.emplace_back();
+						continue;
+					}
+					ret.verified_leaf_hashes.emplace_back(m_torrent_file->files().file_num_blocks(f));
+					std::fill(ret.verified_leaf_hashes.back().begin()
+						, ret.verified_leaf_hashes.back().end(), false);
```

Comment:
instead of a `std::fill()` here, couldn't the `emplace_back()` call just have a second argument of `false`?

Reply:
Right. Good catch.